File: solutions/cpp/space-age/1/space_age.cpp
#include "space_age.h"
#include <bits/stdint-uintn.h>

namespace space_age {
space_age::space_age(long _age) { age = _age; }

double space_age::seconds() const { return age; }

double space_age::on_earth() const { return age / 31557600.0; }

double space_age::on_mercury() const { return age / (0.2408467 * 31557600.0); }
double space_age::on_venus() const { return age / (0.61519726 * 31557600.0); }
double space_age::on_mars() const { return age / (1.8808158 * 31557600.0); }
double space_age::on_jupiter() const { return age / (11.862615 * 31557600.0); }
double space_age::on_saturn() const { return age / (29.447498 * 31557600.0); }
double space_age::on_uranus() const { return age / (84.016846 * 31557600.0); }
double space_age::on_neptune() const { return age / (164.79132 * 31557600.0); }

} // namespace space_age

File: solutions/lua/space-age/1/space-age.lua
local SpaceAge = {}

SpaceAge.seconds = 0
function SpaceAge:new(number, n)
    n = (n or {})
    setmetatable(n, self)
    self.__index = self
    self.seconds = number
    return n
end

function SpaceAge.on_earth()
    local result = math.ceil((SpaceAge.seconds / 315576)) / 100
    return result
end
function SpaceAge.on_mercury()
    local result = math.floor(SpaceAge.on_earth() / 0.002408467) / 100
    return result
end
function SpaceAge.on_mars()
    local result = math.floor(SpaceAge.on_earth() / 0.018808158) / 100
    return result
end
function SpaceAge.on_jupiter()
    local result = math.ceil(SpaceAge.on_earth() / 0.11862615) / 100
    return result
end

function SpaceAge.on_saturn()
    local result = math.floor(SpaceAge.on_earth() / 0.29447498) / 100
    return result
end
function SpaceAge.on_uranus()
    local result = math.ceil(SpaceAge.on_earth() / 0.84016846) / 100
    return result
end
function SpaceAge.on_neptune()
    local result = math.floor(SpaceAge.on_earth() / 1.6479132) / 100
    return result
end

return SpaceAge
